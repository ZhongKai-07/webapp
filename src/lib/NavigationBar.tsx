/* eslint-disable @next/next/no-img-element */
import React from 'react';

import NavigationBarItem from '@/lib/NavigationBarItem';

const NavigationBar: React.FC = () => {
  return (
    <div className='flex justify-between w-full bg-white border-b-2 h-[7.5em]'>
      <div className='ml-[2em] relative flex justify-evenly items-center'>
        <img
          className='float-left reletive w-[10em]'
          width={0}
          height={0}
          alt='apian logo'
          src='/images/Apian_logo_transparent_bg.png'
        />
        <div className='relative block bg-black h-[3.5em] w-[2px] mx-[10px]'></div>
        <img
          className='float-left relative w-[10em]'
          width={0}
          height={0}
          alt='UCL logo'
          src='/images/ucl_logo.jpg'
        />
        <p className='relative w-[15em] text-4xl whitespace-nowrap'>
          Drone Deliveries
        </p>
      </div>
      <div className='mr-[5em] items-center flex relative h-full'>
        <nav className='relative'>
          <ul className='items-center text-2xl flex justify-around gap-[2em] relative'>
            <NavigationBarItem pathname='/' title='Home' />
            <NavigationBarItem
              pathname='/cranespotting'
              title='Crane Spotting'
            />
            <NavigationBarItem pathname='/dropzone' title='Drop Zone' />
          </ul>
        </nav>
      </div>
    </div>
  );
};
export default NavigationBar;
