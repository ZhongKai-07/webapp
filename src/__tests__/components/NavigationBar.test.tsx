import { render } from '@testing-library/react';
import React from 'react';

import NavigationBar from '@/lib/NavigationBar';

describe('NavigationBar component', () => {
  it('should render correctly', () => {
    const { container } = render(<NavigationBar />);
    expect(container).toMatchSnapshot();
  });
});
