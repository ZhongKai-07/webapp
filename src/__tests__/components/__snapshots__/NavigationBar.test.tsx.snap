// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component should render correctly 1`] = `
<div>
  <div
    class="flex justify-between w-full bg-white border-b-2 h-[7.5em]"
  >
    <div
      class="ml-[2em] relative flex justify-evenly items-center"
    >
      <img
        alt="apian logo"
        class="float-left reletive w-[10em]"
        height="0"
        src="/images/Apian_logo_transparent_bg.png"
        width="0"
      />
      <div
        class="relative block bg-black h-[3.5em] w-[2px] mx-[10px]"
      />
      <img
        alt="UCL logo"
        class="float-left relative w-[10em]"
        height="0"
        src="/images/ucl_logo.jpg"
        width="0"
      />
      <p
        class="relative w-[15em] text-4xl whitespace-nowrap"
      >
        Drone Deliveries
      </p>
    </div>
    <div
      class="mr-[5em] items-center flex relative h-full"
    >
      <nav
        class="relative"
      >
        <ul
          class="items-center text-2xl flex justify-around gap-[2em] relative"
        >
          <li>
            <a
              href="/"
            >
              Home
            </a>
            <div
              class="relative -mb-px h-px bg-black w-0"
              style="transition: width 300ms ease;"
            />
          </li>
          <li>
            <a
              href="/cranespotting"
            >
              Crane Spotting
            </a>
            <div
              class="relative -mb-px h-px bg-black w-0"
              style="transition: width 300ms ease;"
            />
          </li>
          <li>
            <a
              href="/dropzone"
            >
              Drop Zone
            </a>
            <div
              class="relative -mb-px h-px bg-black w-0"
              style="transition: width 300ms ease;"
            />
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
